moddingway/database/roles_database.py

```diff
@@ -0,0 +1,52 @@
+from . import DatabaseConnection
+from discord import Role
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+def add_sticky_roles(user_id: int, roles: list[int]):
+    if len(roles) == 0:
+        return
+
+    conn = DatabaseConnection()
+    with conn.get_cursor() as cursor:
+        query = "INSERT INTO roles (userID, roleID) VALUES "
+        query += ",".join([f"({user_id}, {role})" for role in roles])
```

Comment:
Arrays aren't really a thing when dealing with relational databases, it's better to have it as a separate table and connect it to another table with a foreign key (which is what's done here)

Reply:
i'm pretty sure postgres supports arrays. the only reason i want to head this way is because of processing time for the `roles` table. if we have to query multiple rows based on the user id in a database that could potentially have tens/hundreds of thousands of entries, then storing the roles atomically in the `users` table should be good enough to avoid performance issues. i know it doesn't affect the current ticket, but if we're looking to build the firewall in the future, having a huge `roles` table is not unrealistic. iirc the foreign key also adds some overhead cost, but i think it should be an amortized runtime? if we're confident that this won't affect runtime, then this approach is okay.

Reply:
https://stackoverflow.com/questions/20219503/is-it-bad-design-to-use-arrays-within-a-database
From what these answers say, it's generally not a good idea to use them unless we are 100% sure that these won't be used for any other tables. 
https://stackoverflow.com/questions/6843912/what-are-the-proper-use-cases-for-the-postgresql-array-datatype
This gives some examples of what it should be used for. If you're sure you want this I could implement these but I personally don't think it's worth.

Reply:
ok let's stay with this approach for now then. i'm just worried that performance will take a hit later, but we can revamp it if we need to